ait/core/db.py

```diff
@@ -348,13 +463,140 @@ def query(self, query, **kwargs):
+                in the form (packet time as datetime object, Packet object). This
+                provides access to the time field associated with the packet in
+                the database.
+
+            **kwargs**: Additional kwargs are passed to the backend query without
+                modification.
+
+        Returns:
+            An :class:`AITDBResult` with **packets** set to a generator of all
+                packet objects returned by the query in time-sorted order. Otherwise,
+                **errors** will contain any encountered errors.
+
+        Raises:
+            ValueError: If a provided packet type name cannot be located in the
+                telemetry dictionary.
+        '''
+        if packets is not None:
+            tlm_dict = tlm.getDefaultDict()
+            for name in packets:
+                if name not in tlm_dict:
+                    msg = f'Invalid packet name "{name}" provided'
+                    log.error(msg)
+                    raise ValueError(msg)
+            pkt_names = ', '.join(packets)
+        else:
+            pkt_names = ', '.join([f'"{i}"' for i in tlm.getDefaultDict().keys()])
+
+        if start_time is not None:
+            stime = start_time.strftime(dmc.RFC3339_Format)
+        else:
+            stime = dmc.GPS_Epoch.strftime(dmc.RFC3339_Format)
+
+        if end_time is not None:
+            etime = end_time.strftime(dmc.RFC3339_Format)
+        else:
+            etime = dt.datetime.utcnow().strftime(dmc.RFC3339_Format)
+
+        query_string = f"SELECT * FROM {pkt_names} WHERE time >= '{stime}' AND time <= '{etime}'"
+
+        try:
+            db_res = self._query(query_string, **kwargs)
+        except self._backend.exceptions.InfluxDBClientError as e:
+            log.error(f'query_time_range failed with exception: {e}')
+            return AITDBResult(query=query_string, errors=[str(e)])
+
+        def influx_results_gen(db_res, **kwargs):
+            from ait.core import dmc
+
+            res_items = db_res.items()
+            pkt_names = [i[0][0] for i in res_items]
+            pkt_gens = [i[1] for i in res_items]
+
+            for packets in itertools.zip_longest(*pkt_gens, fillvalue=None):
+                pkt_conv = [
+                    (
+                        dt.datetime.strptime(d['time'], dmc.RFC3339_Format),
+                        InfluxDBBackend.create_packet_from_result(pkt_names[i], d)
+                    )
+                    for i, d in enumerate(packets)
+                    if d is not None
+                ]
+
+                pkt_conv.sort(key=lambda x: x[0])
+
+                for t, pkt in pkt_conv:
+                    if kwargs.get('yield_packet_time', False):
+                        yield (t, pkt)
+                    else:
+                        yield pkt
+
+        return AITDBResult(
+            query=query_string,
+            packets=influx_results_gen(db_res, **kwargs)
```

Comment:
@MJJoyce Just wondering if there would be any value in having the return object contain the raw results along with the Packet generator?  I don't have a strong opinion either way.

Reply:
Hmm I don't think so here. The `query_packets` API is meant to give you back `tlm.Packet`s. If users just want raw query results they can do that through the `query` endpoint. Since running through the raw results would require you to be familiar with the DB result format I don't think it's too much of a stretch to require they also be familiar with the query syntax.

Thoughts?